src/_posts/languages/php/2000-01-01-dependencies.md

```diff
@@ -0,0 +1,105 @@
+title: Managing Dependencies
+nav: Managing Dependencies
+modified_at: 2023-06-02 16:00:00
+tags: php
+index: 4
+---
+
+[Composer](https://getcomposer.org) is the official dependency manager for PHP.
+It allows to manage (download, install, upgrade, configure, and remove) the
+libraries a project depends on.
+
+If you want the platform to use Composer to manage your dependencies, make sure
+to include both the `composer.json` and the `composer.lock` files in your
+codebase.
+
+## Declaring Dependencies
+
+The dependencies required by your application must be declared in a file called
+`composer.json`, stored at the root of your codebase. The format is described
+in the [Composer documentation](https://getcomposer.org/doc/01-basic-usage.md).
+
+Once your dependencies have been defined and declared, their versions must be
+frozen to ensure a precise version of the application will always be deployed
+with the same compatible set of Composer packages. This allows for better
+reproducibility and consistency across environments. These versions are written
+in a file called `composer.lock`, also stored at the root of your project.
+
+The `composer.lock` file is generated by Composer itself:
+
+```bash
+$ composer install
+```
+
+To upgrade a dependency, run the following command (in the example below, we
+ask Composer to upgrade `slim`):
+
+```bash
+$ composer update slim/slim
+```
+
+In some circumstances, it can be convenient to add either the
+`--ignore-platform-req=` or the `--ignore-platform-reqs` flag to the above
+commands.
+
+After each command, the `composer.lock` file is automatically updated. Don't
+forget to commit the modifications!
+
+{% note %}
+    Both the `scalingo.json` and `scalingo.lock` files are _required_ for the
+    platform to handle your dependencies.
+{% endnote %}
+
+### Managing Private Dependencies
+
+If you want to install a private dependency, you need to define the
+`COMPOSER_AUTH` environment variable, as specified in
+[the Composer documentation](https://getcomposer.org/doc/03-cli.md#composer-auth).
+For instance, for a private dependency hosted on GitHub, the `COMPOSER_AUTH`
+environment variable should contain:
+
+```json
+{
+  "github-oauth": {
+    "github.com": "MY-TOKEN"
+  }
+}
+```
+
+`MY-TOKEN` must be replaced with a valid token of your GitHub account.
+
+## Specifying the Composer Version
+
+You can select the Composer version to install by specifying it in your
+`composer.json`:
+
+```json
+{
+  "extra": {
+    "paas": {
+      "engines": {
+        "composer": "2.x"
+      }
+    }
+  }
+}
+```
+
+Scalingo currently supports the following versions of Composer:
+
+- `2.2.18`
+- `2.3.10`
+- `2.4.4`
+- `2.5.7`
+
+## Working with Composer Environments
+
+By default, Scalingo considers that your application runs in *production* mode.
+This means that `composer install` automatically runs with the `--no-dev` flag,
+and, as a result, won't install the development dependencies of your
+application, if any.
```

Comment:
suggestion(non-blocking): this sentence is really long, hence hard to read. Maybe split it into multiple, smaller, sentences?

```suggestion
This means that `composer install` automatically runs with the `--no-dev` flag.
As a result, Composer won't install the development dependencies of your
application, if any.
```

Reply:
You're right, applied in https://github.com/Scalingo/documentation/pull/2139/commits/829c67bb4f86ff17f20dea99282b1cacf766833f